iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/serviceclient/ExportImportTests.java

```diff
@@ -160,8 +213,18 @@ public void export_import_e2e() throws Exception
         {
             try
             {
-                exportJob = registryManager.exportDevices(containerSasUri, excludeKeys);
-                exportJobScheduled = true;
+                if(storageAuthenticationType.isPresent())
+                {
+                    JobProperties exportJobProperties = JobProperties.createForExportJob(containerSasUri, excludeKeys, storageAuthenticationType.get());
+                    exportJob = registryManager.exportDevices(exportJobProperties);
```

Comment:
In my experience running several tests at once now, I've run into an JobQuotaExceededException. I thought of different ways of handling this, including using a semaphore, but that would only prevent tests in the same process from running at the same time.

I eventually landed on just guarding against that exception and retrying a certain number of times.

You'll want to do the same, otherwise these tests will occasionally fail and be considered flaky.

Reply:
I see a catch for IotHubTooManyDevicesException, which seems like a weird one to throw, but maybe in Java it is a differently named exception.

If that is the same, you have it covered, although I think a 30 second sleep is a bit much. We don't need tests taking longer than they need to. In my test, I loop for a max of 30 times, sleeping 1 second each time. If I need to increase the iteration count because of failures, I will, but I'd rather know the jobs are taking a long time or that we have too many concurrent ones before I let tests run too long.

Reply:
yes you are right, it's a same exception, error code: 403. I will change sleep time .